napytau/gui/gui_app.py

```diff
@@ -0,0 +1,478 @@
+from typing import List, Tuple
+
+import tkinter as tk
+from tkinter import Canvas
+from tkinter import filedialog
+
+import customtkinter
+from matplotlib.figure import Figure
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg
+
+from napytau.cli.cli_arguments import CLIArguments
+from napytau.core.logic_mockup import logic
+from tkinter import Menu
+
+from napytau.gui.checkbox_datapoint import CheckboxDataPoint
+
+customtkinter.set_appearance_mode(
+    "System"
+)  # Modes: "System" (standard), "Dark", "Light"
+customtkinter.set_default_color_theme(
+    "blue"
+)  # Themes: "blue" (standard), "green", "dark-blue"
+
+
+# Modes: "System" (standard), "Dark", "Light"
+customtkinter.set_appearance_mode("System")
+# Themes: "blue" (standard), "green", "dark-blue"
+customtkinter.set_default_color_theme("blue")
+
+class GUIApp(customtkinter.CTk):
+    def __init__(self) -> None:
+        """
+        Constructor for the GUIApp, initializes the GUI.
+        """
+        super().__init__()
+
+        # Datapoints
+        self.datapoints: List[Tuple[float, float]]=[]
+        self.datapoints_for_fitting: List[CheckboxDataPoint]=[]
+        self.datapoints_for_calculation: List[CheckboxDataPoint]=[]
+
+        # values
+        self.tau = tk.IntVar()
+        self.tau.set(2)
+
+        # configure window
+        self.title("NaPyTau")
+        self.geometry("1366x768")
+
+        """
+        Configure grid. Current Layout:
+        Three rows, two columns with
+        - Graph from row 0 to 1, column 0
+        - Checkboxes in row 0, column 1
+        - Control area in row 1, column 1
+        - Information area in row 2, column 0 to 1
+        """
+        self.grid_rowconfigure((0, 2), weight=1)  # Three rows
+        self.grid_columnconfigure((0, 1), weight=1)  # Two columns
+
+        # Weights are adjusted
+        self.grid_rowconfigure(0, weight=10)
+        self.grid_rowconfigure(1, weight=1)
+        self.grid_rowconfigure(2, weight=2)
+        self.grid_columnconfigure(0, weight=8)
+        self.grid_columnconfigure(1, weight=1)
+
+        # Initialize menu bar
+        self._init_menu_bar()
+
+        # Initialize graph
+        self._init_graph()
+
+        # Initialize Checkboxes with datapoints
+        self._init_datapoint_checkboxes()
+
+        # Initialize control area
+        self._init_control_area()
+
+        # Initialize information area
+        self._init_information_area()
```

Comment:
Area is a really general term, as is information. We might consider renaming this to `control_panel` and `log_list`.

Reply:
I like these suggestions and noted them. But I think this should be part of separate issues.